test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs

```diff
@@ -50,8 +51,42 @@ public override void Dispose()
         }
 
         [Fact]
+        public void ProcessMarkdownTocWithComplexHrefShouldSucceed()
+        {
+            var fileName = "#ctor";
+            var href = HttpUtility.UrlEncode(fileName);
+            var content = $@"
+#[Constructor]({href}.md)
+";
+            var file = _fileCreator.CreateFile(string.Empty, FileType.MarkdownContent, fileNameWithoutExtension: fileName);
+            var toc = _fileCreator.CreateFile(content, FileType.MarkdownToc);
+            FileCollection files = new FileCollection(_inputFolder);
+            files.Add(DocumentType.Article, new[] { toc, file });
+            BuildDocument(files);
+
+            var outputRawModelPath = Path.GetFullPath(Path.Combine(_outputFolder, Path.ChangeExtension(toc, RawModelFileExtension)));
+            Assert.True(File.Exists(outputRawModelPath));
+            var model = JsonUtility.Deserialize<TocItemViewModel>(outputRawModelPath);
+            var expectedModel = new TocItemViewModel
+            {
+                Items = new TocViewModel
+                {
+                    new TocItemViewModel
+                    {
+                        Name = "Constructor",
+                        Href = $"{href}.md",
+                        TopicHref = $"{href}.md",
+                    }
+                }
+            };
+
+            AssertTocEqual(expectedModel, model);
+        }
+
+        [Fact]
         public void ProcessMarkdownTocWithAbsoluteHrefShouldSucceed()
         {
+            var file1 = _fileCreator.CreateFile(string.Empty, FileType.MarkdownContent, "a");
```

Comment:
....

---
In reply to: [96348009](https://github.com/dotnet/docfx/pull/1188#discussion_r96348009) [](ancestors = 96348009)